### ISSUE TITLE
Connector does not support fault tolerant execution

### ISSUE THREAD

**Issue by aleksanderlech** [nineinchnick/trino-openapi#132]
Hey,

We are running connector version 1.63 together with Trino 459 and have configured the following setting:

```yaml
additionalConfigProperties:
  - retry-policy=TASK
```

After that the connector fails for any query.

```java
java.lang.UnsupportedOperationException: This connector does not provide memory accounting capabilities for ConnectorSplit
	at io.trino.spi.connector.ConnectorSplit.getRetainedSizeInBytes(ConnectorSplit.java:58)
	at io.trino.metadata.Split.getRetainedSizeInBytes(Split.java:95)
	at io.airlift.slice.SizeOf.estimatedSizeOf(SizeOf.java:200)
	at io.trino.execution.scheduler.faulttolerant.SplitsMapping.lambda$getRetainedSizeInBytes$1(SplitsMapping.java:120)
	at io.airlift.slice.SizeOf.estimatedSizeOf(SizeOf.java:241)
	at io.trino.execution.scheduler.faulttolerant.SplitsMapping.lambda$getRetainedSizeInBytes$2(SplitsMapping.java:117)
	at io.airlift.slice.SizeOf.estimatedSizeOf(SizeOf.java:241)
	at io.trino.execution.scheduler.faulttolerant.SplitsMapping.getRetainedSizeInBytes(SplitsMapping.java:114)
	at io.trino.execution.scheduler.faulttolerant.TaskDescriptor.getRetainedSizeInBytes(TaskDescriptor.java:91)
	at io.trino.execution.scheduler.faulttolerant.TaskDescriptorStorage$TaskDescriptors.put(TaskDescriptorStorage.java:288)
	at io.trino.execution.scheduler.faulttolerant.TaskDescriptorStorage.put(TaskDescriptorStorage.java:118)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$StagePartition.seal(EventDrivenFaultTolerantQueryScheduler.java:2751)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$StageExecution.sealPartition(EventDrivenFaultTolerantQueryScheduler.java:2110)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$Scheduler.lambda$onSplitAssignment$29(EventDrivenFaultTolerantQueryScheduler.java:1844)
	at com.google.common.primitives.ImmutableIntArray.forEach(ImmutableIntArray.java:415)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$Scheduler.onSplitAssignment(EventDrivenFaultTolerantQueryScheduler.java:1843)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$Scheduler.onSplitAssignment(EventDrivenFaultTolerantQueryScheduler.java:666)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$SplitAssignmentEvent.accept(EventDrivenFaultTolerantQueryScheduler.java:3426)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$Scheduler.processEvents(EventDrivenFaultTolerantQueryScheduler.java:919)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenFaultTolerantQueryScheduler$Scheduler.run(EventDrivenFaultTolerantQueryScheduler.java:839)
	at io.trino.$gen.Trino_459____20241014_080640_2.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
```

Is there an easy way to add this fault tolerance capabilities?

**Comment by nineinchnick** [comment]
I've never added support for FTE in any connector, yet. It should be fairly easy to add for reads.

**Comment by aleksanderlech** [comment]
I know its more as a feature request for the future than an actual bug :)

**Comment by vcknorket** [comment]
Can this be added for reads? If its easy? Would be great to have and so much appreciate it

**Comment by nineinchnick** [comment]
Fixed in 9b012d23e4a59af18c6d44af8dc00bbf7d3a9b45
